R/shiny-user.R

```diff
@@ -0,0 +1,18 @@
+#' Get Crunch user details and confirm authentication
+#'
+#' `shinyUser()` returns information about the current user of the Shiny app. This
+#' is useful if you want to change the behavior of the app depending on who is
+#' viewing the app. You can access elements of this record with Crunch functions
+#' like `name()` or `email().`
+#'
+#' `checkAuthentication()` is an alias for `shinyUser()` for when you only care
+#' about ensuring that the current user is authenticated with Crunch.
+#'
+#' @return A user record if the user is logged in, otherwise a character vector or error
+#' @export
+#' @importFrom crunch me
+shinyUser <- function () .buildReactiveExpr("me")
+
+#' @rdname shinyUser
+#' @export
+checkAuthentication <- shinyUser
```

Comment:
Thinking about this some more, I'm wondering if instead we can do

```
checkAuthentication <- function () {
    token <- reactive(input$token)()
    if (is.null(token) {
        stop("You are not authenticated with Crunch. Please go to https://app.crunch.io and log in.", call.=FALSE)
    }
    invisible(TRUE)
}
```

It only checks whether you have an auth cookie set, not whether it's currently valid, but it doesn't require a network request. That is probably sufficient.

Reply:
The only way I seem to be able to get this to work is by passing `input` down to to checkAuthentication. If I just call `checkAuthentication()` in the server function I get an `object input not found` error. It looks like `checkAuthentication` doesn't have access to the server function's scope, but I'm really stumped as to how that's possible.

Reply:
Yeah there's some dark magic in that `reactive()` thing. Surely there's a way to make it work though.